tools/testsys-config/src/lib.rs

```diff
@@ -0,0 +1,364 @@
+    /// The role that should be assumed for this particular variant
+    pub agent_role: Option<String>,
+    /// The custom images used for conformance testing
+    pub conformance_image: Option<String>,
+    /// The custom registry used for conformance testing
+    pub conformance_registry: Option<String>,
+}
+
+impl GenericVariantConfig {
+    /// Overwrite the unset values of `self` with the set values of `other`
+    fn merge(self, other: Self) -> Self {
+        let cluster_names = if self.cluster_names.is_empty() {
+            other.cluster_names
+        } else {
+            self.cluster_names
+        };
+
+        let secrets = if self.secrets.is_empty() {
+            other.secrets
+        } else {
+            self.secrets
+        };
+
+        Self {
+            cluster_names,
+            instance_type: self.instance_type.or(other.instance_type),
+            secrets,
+            agent_role: self.agent_role.or(other.agent_role),
+            conformance_image: self.conformance_image.or(other.conformance_image),
+            conformance_registry: self.conformance_registry.or(other.conformance_registry),
+        }
+    }
+}
+
+impl From<GenericVariantConfig> for AwsK8sVariantConfig {
+    fn from(val: GenericVariantConfig) -> Self {
+        Self {
+            cluster_names: val.cluster_names,
+            instance_type: val.instance_type,
+            secrets: val.secrets,
+            assume_role: val.agent_role,
+            kube_conformance_image: val.conformance_image,
+            e2e_repo_registry: val.conformance_registry,
+        }
+    }
+}
+
+impl From<GenericVariantConfig> for AwsEcsVariantConfig {
+    fn from(val: GenericVariantConfig) -> Self {
+        Self {
+            cluster_names: val.cluster_names,
+            instance_type: val.instance_type,
+            secrets: val.secrets,
+            assume_role: val.agent_role,
+        }
+    }
+}
+
+/// Fill in the templated cluster name with `arch` and `variant`.
+pub fn rendered_cluster_name<S1, S2>(cluster_name: String, arch: S1, variant: S2) -> Result<String>
+where
+    S1: Into<String>,
+    S2: Into<String>,
+{
+    let mut cluster_template = Handlebars::new();
+    cluster_template.register_template_string("cluster_name", cluster_name)?;
+    Ok(cluster_template.render(
+        "cluster_name",
+        &btreemap! {"arch".to_string() => arch.into(), "variant".to_string() => variant.into()},
+    )?)
+}
+
+#[derive(Debug, Default, Deserialize, Serialize, PartialEq, Eq, Clone)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "kebab-case")]
+pub struct TestsysImages {
+    pub eks_resource_agent_image: Option<String>,
+    pub ecs_resource_agent_image: Option<String>,
+    pub ec2_resource_agent_image: Option<String>,
+    pub sonobuoy_test_agent_image: Option<String>,
+    pub ecs_test_agent_image: Option<String>,
+    pub migration_test_agent_image: Option<String>,
+    pub testsys_agent_pull_secret: Option<String>,
```

Comment:
Can I have one agent container that is published in a different place? Or would I need to publish all of these containers to the same container registry?

Reply:
They can be in different ones. There is a note about it in `Test.toml.example`.